### ISSUE TITLE
Duplicate Key Warn/Error in Stage Collection

### ISSUE THREAD

**Issue by nicholsn** [opencb/opencga#766]
I'm seeing the warning below when running the annotation pipeline in 1.3.0-rc for multiple studies with the following command:

```
$OPENCGA_BIN/opencga-analysis.sh variant index \
    --file $WGS_FILE \
    --outdir $WORK_DIR \
    --calculate-stats \
    --annotate \
    --annotator cellbase_db_adaptor \
    -s $STUDY \
    -S $TOKEN
```
Is there anything to be concerned about with this warning?
```
2017-12-24 22:32:15 [pool-2-thread-7] WARN  MongoDBVariantStageLoader:189 - Catch error : BulkWriteError{index=83, code=11000, message='E11000 duplicate key error collection: opencga_user_project.stage index: _id_ dup key: { : " 2:  91519059:G:C" }', details={ }}
2017-12-24 22:32:15 [pool-2-thread-7] WARN  MongoDBVariantStageLoader:190 - DupKey exception inserting ' 2:  91519059:G:C'. Retry!
```

Furthermore, the pipeline throws and error and fails when it comes across one of these duplicate keys that has some binary encoding in the key. 

This may be a separate issue, but do you have any ideas why there would be binary characters (e.g., `{ : " 2:  91523836:��z"wy#|��VqMI�f�pn�lu:*" }`) in the keys?

```
2017-12-24 22:32:21 [pool-2-thread-7] ERROR MongoDBVariantStageLoader:192 - WriteError with code 11000 does not match with the pattern ^.*dup key: \{ : "([^"]*)" \}$
2017-12-24 22:32:21 [pool-2-thread-7] ERROR ParallelTaskRunner:830 - Error writing batch 50413
com.mongodb.MongoBulkWriteException: Bulk write operation error on server localhost:27017. Write errors: [BulkWriteError{index=45, code=11000, message='E11000 duplicate key error collection: opencga_user_project.stage index: _id_ dup key: { : " 2:  91523836:��z"wy#|��VqMI�f�pn�lu:*" }', details={ }}].
	at com.mongodb.connection.BulkWriteBatchCombiner.getError(BulkWriteBatchCombiner.java:176)
	at com.mongodb.connection.BulkWriteBatchCombiner.throwOnError(BulkWriteBatchCombiner.java:205)
	at com.mongodb.connection.BulkWriteBatchCombiner.getResult(BulkWriteBatchCombiner.java:146)
	at com.mongodb.operation.MixedBulkWriteOperation$1.call(MixedBulkWriteOperation.java:188)
	at com.mongodb.operation.MixedBulkWriteOperation$1.call(MixedBulkWriteOperation.java:168)
	at com.mongodb.operation.OperationHelper.withConnectionSource(OperationHelper.java:422)
	at com.mongodb.operation.OperationHelper.withConnection(OperationHelper.java:413)
	at com.mongodb.operation.MixedBulkWriteOperation.execute(MixedBulkWriteOperation.java:168)
	at com.mongodb.operation.MixedBulkWriteOperation.execute(MixedBulkWriteOperation.java:74)
	at com.mongodb.Mongo.execute(Mongo.java:845)
	at com.mongodb.Mongo$2.execute(Mongo.java:828)
	at com.mongodb.MongoCollectionImpl.bulkWrite(MongoCollectionImpl.java:301)
	at org.opencb.commons.datastore.mongodb.MongoDBNativeQuery.update(MongoDBNativeQuery.java:259)
	at org.opencb.commons.datastore.mongodb.MongoDBCollection.update(MongoDBCollection.java:404)
	at org.opencb.opencga.storage.mongodb.variant.load.stage.MongoDBVariantStageLoader.updateMongo(MongoDBVariantStageLoader.java:168)
	at org.opencb.opencga.storage.mongodb.variant.load.stage.MongoDBVariantStageLoader.insert(MongoDBVariantStageLoader.java:102)
	at org.opencb.opencga.storage.mongodb.variant.load.stage.MongoDBVariantStageLoader.write(MongoDBVariantStageLoader.java:93)
	at org.opencb.commons.run.ParallelTaskRunner$WriterRunnable.call(ParallelTaskRunner.java:828)
	at org.opencb.commons.run.ParallelTaskRunner$WriterRunnable.call(ParallelTaskRunner.java:811)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2017-12-24 22:32:21 [pool-2-thread-7] WARN  ParallelTaskRunner:836 - Abort writing thread on fail
2017-12-24 22:52:21 [main] WARN  ParallelTaskRunner:446 - Catch interrupted exception!
java.util.concurrent.TimeoutException: Queue got stuck with 12 items!!!
```

**Comment by imedina** [comment]
Many thanks for spending the time to write this issue. A couple of comments here.

1. the `DupKey exception` warning is not an issue, this happens when loading files in parallel, is a controlled exception and it is handled properly, this is why it is a _warning_. Also, do not worry about the binary characters, this is an internal codification for MongoDB id for structural variants.
2. the error you had experience is the first time has happened in 2 years and thousands of files indexed, actually it needs that three different unlikely situations happen simultaneously.

The good news is that @j-coll has solved this issue today and it is already merged into **1.3.0-rc** and **develop**, as you can see in the commit above.

Thanks again!

**Comment by nicholsn** [comment]
@imedina, thanks for the update - I will try to rerun the failed pipeline and report back any issues
